src/main/filesystem/watcher.js

```diff
@@ -182,9 +264,48 @@ class Watcher {
     }
   }
 
-  clear () {
-    Object.keys(this.watchers).forEach(id => this.watchers[id].watcher.close())
+  close () {
+    Object.keys(this.watchers).forEach(id => this.watchers[id].close())
     this.watchers = {}
+    this._ignoreChangeEvents = []
+  }
+
+  /**
+   * Ignore the next changed event within a certain time for the current file and window.
+   *
+   * NOTE: Only valid for files and "add"/"change" event!
+   *
+   * @param {number} windowId The window id.
+   * @param {string} pathname The path to ignore.
+   * @param {number} [duration] The duration in ms to ignore the changed event.
+   */
+  ignoreChangedEvent (windowId, pathname, duration=WATCHER_STABILITY_THRESHOLD + WATCHER_STABILITY_POLL_INTERVAL + 1000) {
```

Comment:
How about use `disableChangedEvent(windowId, pathname)` and `enableChangedEvent(windowId, pathname)` to instead of `ignoreChangeEvent`? Because I think the duration is not easy to control.

Reply:
Well, the core problem is that the time when we have to re-enable the events is unknown. We know that we changed a file but cannot say when the associated event is fired (due CPU/IO scheduling and the writer may write continuously for a short time (e.g. file stream)). We know that the add/change event is emitted after no manipulation is done to the file after the watcher stability threshold. Meaning the minimum time is 1000ms after a event is emitted from the OS and the file is checks in intervals of 100ms for size changes.

So I assumed that we have to wait for the stability threshold + 1s to be sure that we don't receive the changed notification. I guess that before this change the main process was blocked and has never received the change notification. Checking for file size or last modification time is also problematic because `fs.stat` may read the wrong information when the content is not flushed on disk. But yes, this needs an improvement but I don't really know how.

Reply:
Use this solution first, and then think of a better solution and then replace it.